### PR TITLE
[Enhancement] Add a featured product button

### LINKED ISSUE #39679: [Enhancement] Add a featured product button
We need a way to mark a product as featured. I thought we could have a button in the name field and use a star icon to convey its meaning.

**Design**

AkVNGImLgSqCObTQ3idVn7-fi-1005_134951

AkVNGImLgSqCObTQ3idVn7-fi-7001_308034 ( updated design link )

image

**Acceptance criteria**

- [x] There's a star button on the right side of the name field. It's centered vertically.
- [x] Clicking it marks the product as featured.
- [x] When hovered, we show a tooltip that reads: `Mark as featured`
- [x] When clicked, the icon changes from `star-outline` to `star-filled`. Clicking it again reverts to the original icon.

### PR DESCRIPTION
**Pull request by mdperez86**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes #39679

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Make sure feature `New product editor` is enabled under `/wp-admin/admin.php?page=wc-settings&tab=advanced&section=features`
2. Go to `Products` -> `Add new` from the left side menu.
3. As part of the `Name` field a new star icon button should be shown on the right side.
4. Clicking it marks the product as featured.
5. When hovered, we show a tooltip that reads: `Mark as featured` 
image

6. When clicked, the icon changes from an outlined star to a filled star. Clicking it again reverts to the original icon.
7. Saving the product should persist it as `featured` product if the star icon is filled.

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

</details>

### CODE REVIEW

File: packages/js/product-editor/src/blocks/product-fields/name/edit.tsx [hunk 1]
```diff
@@ -156,6 +158,36 @@ export function Edit( {
...
+		useProductEntityProp< boolean >( 'featured' );
+
+	function handleSuffixClick() {
+		setFeatured( ! featured );
+	}
+
+	function renderFeaturedSuffix() {
+		return (
+			<Tooltip
+				text={ __( 'Mark as featured', 'woocommerce' ) }
```
Comment (louwie17) [inline packages/js/product-editor/src/blocks/product-fields/name/edit.tsx]: I am ok leaving this as is, but given we do use `Unmark as featured` for the aria label below, I was wondering if we should update this to say `Unmark` when it is selected?

**Review by louwie17 — APPROVED** [review]
LGTM 🚀 Tested well, nice work!

File: packages/js/product-editor/src/blocks/product-fields/name/edit.tsx (hunk 1 above)
Comment (mdperez86) [reply packages/js/product-editor/src/blocks/product-fields/name/edit.tsx]: Done -> https://github.com/woocommerce/woocommerce/pull/43664/commits/e0b8bc426ec441760c51619ccc79772d714b595c

**Review by louwie17 — APPROVED** [review]
LGTM 🚀 Thanks for the update.

**Pull request merged**